90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -447,20 +480,17 @@ kexec_kernel() {
   cli_args="$( load_be_cmdline "${fs}" )"
   root_prefix="$( find_root_prefix "${fs}" "${mnt}" )"
 
-  # restore kernel log level just before we kexec
-  # shellcheck disable=SC2154
-  if [ -n "${PRINTK}" ] ; then
-    echo "${PRINTK}" > /proc/sys/kernel/printk
-    zdebug "restored kernel log level to ${PRINTK}"
-  fi
-
   if ! output="$( kexec -l "${mnt}${kernel}" \
     --initrd="${mnt}${initramfs}" \
     --command-line="${root_prefix}${fs} ${cli_args}" 2>&1 )"
   then
     zerror "unable to load ${mnt}${kernel} and ${mnt}${initramfs} into memory"
     zerror "${output}"
     umount "${mnt}"
+    color=red delay=10 timed_prompt "Unable to load kernel or initramfs into memory" \
+      "${mnt}${kernel}" \
+      "${mnt}${initramfs}"
```

Comment:
Would it be reasonable to drop this BE from the menu after this message? There's no reason to expect a later attempt to load will suddenly succeed.

Reply:
The only thing I can think of is that maybe you'd chroot into the BE and fix whatever is wrong, but that's a bit of a stretch. I'm okay with discarding the BE after a failed kexec load, or kexec execute.

Reply:
There's no readily available mechanism to remove what is (by current tests) a valid BE. We'll need to establish how to blacklist a BE at runtime in a new PR/branch.